### PR TITLE
Kortex simulations

### PR DESCRIPTION
**Pull request by destogl**
## Notes
- URDF files are extended to support Gazebo and Ignition simulations
- new file kinova.urdf.xacro is added as generic "entry-point" toward URDFs (gen3.xacro can probably be deleted)
- gen3_macro.xacro is completely changed because it had windows line endings﻿. They are converted to linux style.

## Usage:

Checkout repos from `simulations.repos` file.

After launching simulation you can test the robot's behavior using test script for joint trajectory controller.
Package ros2_control_demos is needed for this.

```
ros2 launch kortex2_bringup test_joint_trajectory_controller.launch.py
```

#### Ignition:

```
ros2 launch kortex2_bringup kortex_sim_control.launch.py
```

### Gazebo:
Note: I created this as a first test because the setup is almost the same and I know how to setup Gazebo and to check that all URDF file are configured properly before testing Ignition.

```
ros2 launch kortex2_bringup kortex_sim_control.launch.py gazebo_sim:=true ignition_sim:=false
```

## Note 2
@livanov if we are now targeting to setup our scenario we should probably create a new URDF. Although, the first test with camera can be placed in `kinova.urdf.xacro`.

### CODE REVIEW

**Review by livanov93 — CHANGES_REQUESTED** [review]
@destogl I wasn't able to build this until I added [https://github.com/ignitionrobotics/ros_ign/tree/ros2](https://github.com/ignitionrobotics/ros_ign/tree/ros2) into workspace and have built it from source.

**Comment by livanov93** [comment]
@destogl Are we adding `ros_ign` as dependency?

File: simulations.repos [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+  gazebo_ros2_control:
+    type: git
+    url: https://github.com/ros-simulation/gazebo_ros2_control.git
+    version: master
+
+  ign_ros2_control:  # ahcorde/galactic/ign_ros2_control
+    type: git
+    url: https://github.com/ignitionrobotics/ign_ros2_control.git
+    version: cd8c9c0569bfc68a0102cb9b0bb34da00e92b5fb  # Migration from Foxy to Galactic (#10)
```
Comment (destogl) [inline simulations.repos]: Has to be fixed for now because that is an issue on development branch which I didn't debug yet.

**Review by livanov93 — APPROVED** [review]
Great thanks!
I have tested it locally and within gui-based docker and the gazebo and ignition simulation works flawlessly.

We just need to add additional documentation into main readme on how to start the simulation,

**Pull request merged**